src/coreclr/jit/emit.cpp

```diff
@@ -50,6 +50,11 @@ int emitLocation::GetInsNum() const
     return emitGetInsNumFromCodePos(codePos);
 }
 
+void emitLocation::SetInsNum(int insNum)
```

Comment:
I'm worried about this; emitLocations have up until this point been captured and never updated, so I'm worried about adding an updater.

Reply:
The locations store the ig and offset within it. It will assert later if it finds that the offset is larger than the ig instruction count. We could avoid updating the ranges by softening the check and assuming that if the count in the location is larger than the ig then the ig has been truncated. I'm quite aware that this is the first code outside the nop alignment that edits this late in the pipeline and that it could be an exemplar for doing the right things not the easy things.

I chose to update the locations to make sure that I wasn't making these changes work by softening assertions but by making sure they were correct and all consequences of my changes were observed. Now I have it working and the full scope of the removal can been seen you can make an informed decision about whether updating or another option is the better choice.

Reply:
> I'm quite aware that this is the first code outside the nop alignment that edits this late in the pipeline

Can you point to the code that does similar thing for loop alignment?

Reply:
I thought loop alignment added nops of varied lengths after everything else had been done.
```
void CodeGen::genGenerateMachineCode()
{
...
    genCodeForBBlist();
...
    if (GetEmitter()->emitRemoveJumpToNextInst())
    {
        genUpdateLiveRangesForTruncatedIGs();
    }
    genGeneratePrologsAndEpilogs();

    GetEmitter()->emitJumpDistBind();

#if FEATURE_LOOP_ALIGN
    GetEmitter()->emitLoopAlignAdjustments();
#endif

    /* The code is now complete and final; it should not change after this. */
}
``` 
And that alignment was flagged in `genCodeForBBlist()` but instructions added in `emitLoopAlignAdjustments()`. I may have got that wrong.